Update 2020-08-21-unity-csharp-games-security.md (#640)
oktadeveloper/okta-blog: _source/_posts/2020-08-21-unity-csharp-games-security.md

- Example code: [Native](https://github.com/nickgamb/Okta_Unity/blob/master/PanelManager.cs) / [OAuth](https://github.com/nickgamb/Okta_Unity/blob/master/OktaOAuth2.cs).
- The Okta SDK Unity Package can be downloaded [here.](https://github.com/nickgamb/Okta_Unity/blob/master/OktaSDK.unitypackage).

## Videos

Follow along with our companion videos.

NATIVE: [https://youtu.be/9C5xaaLt7Q4](https://youtu.be/9C5xaaLt7Q4)
<div style="text-align: center">
<iframe width="600" height="338" src="https://www.youtube.com/embed/9C5xaaLt7Q4" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
</div>

OAUTH: [https://youtu.be/ciDl0-Wtfvo](https://youtu.be/ciDl0-Wtfvo)
<div style="text-align: center">
<iframe width="600" height="338" src="https://www.youtube.com/embed/ciDl0-Wtfvo" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
</div>

> 

Reviewer: Why remove?

Author: removed because they show the old UI and no CLI

Author: Added back in after chatting with @mraible in slack.

Click `MenuManager` and make sure it is enabled in the inspector. Find the `PanelManager` script component attached to the `MenuManager` object and double click to edit. This will open up the primary IDE configured in Unity.

## Okta CLI

> {% include setup/cli.md type="web" loginRedirectUri="http://127.0.0.1:51772" logoutRedirectUri="http://127.0.0.1:51772"%}

Reviewer: ```suggestion
{% include setup/cli.md type="web" loginRedirectUri="http://127.0.0.1:51772" logoutRedirectUri="http://127.0.0.1:51772" %}
```

1. Drag the `AuthMenu` object from the project hierarchy to the `authMenu` property in the inspector.
2. Drag the InputField from the `Username` object in the project hierarchy to the `Username` property in the inspector.
3. Drag the InputField from the `Password` object in the project hierarchy to the `Password` property in the inspector.

> 4. Replace `{yourOktaDomain}` with the domain of the Okta developer org created by the CLI.

Reviewer: ```suggestion
4. Replace `{yourOktaDomain}` with the domain of your Okta developer org.
```

1. Drag the `AuthMenu` object from the project hierarchy to the `authMenu` property in the inspector.
2. Drag the InputField from the `Username` object in the project hierarchy to the `Username` property in the inspector.
3. Drag the InputField from the `Password` object in the project hierarchy to the `Password` property in the inspector.

> 4. Replace `{yourOktaDomain}` with the domain of your Okta developer org.

Reviewer: ```suggestion
4. Replace `{yourOktaDomain}` with your Okta domain.
```

{% img blog/unity-csharp-security/13-unity-gameobject.png alt:"Unity import package" width:"1000" %}{: .center-image }

From the Okta CLI `.okta.env`, copy the **OKTA_OAUTH2_CLIENT_ID** and **OKTA_OAUTH2_CLIENT_SECRET** into the **Client ID** and **Client Secret** variables in the inspector for the `OktaOAuth2` GameObject in Unity.

{% img blog/unity-csharp-security/18-unity-clientidsecret.png alt:"Okta add web app" width:"1000" %}{: .center-image }

> For the three endpoint URL's, go back to Okta. In the menu click **API** > **Authorization Servers**.

Reviewer: ```suggestion
For the three endpoint URLs, go back to Okta (use `okta login` to get a link in your terminal). In the menu click **Security** > **API** > **Authorization Servers**.
```
